src/reducers/test/testNameReducer.js

```diff
@@ -0,0 +1,64 @@
+export default function (state = {}, action = null) {
+
+  switch(action.type) {
+
+    case "TEST_NAME_CHANGED": {
+      const newState = {
+        name: action.payload,
+        isVisible: true
+      };
+      return newState;
+    }
+
+    case "TEST_NAME_CLEARED": {
+      let newState;
+      if(action.payload === true) {
+        newState = {
```

Comment:
I would assign name on condition as isVisible is same in both cases

Reply:
Done.